src/Malawi_cor_labels_GEE script.js

```diff
@@ -0,0 +1,342 @@
+
+var EPA_color = empty.paint({
+featureCollection: Malawi_EPA,
+color: 1,
+width: 2
+});
+
+var TA_color = empty.paint({
+featureCollection: Malawi_TA,
+color: 1,
+width: 2
+});
+
+//Display the cropland map on the left panel
+var vis_crop = {min: 0, max: 1.0, palette: palette.slice(0,-2)};
+leftMap.addLayer(malawi_binary, vis_crop, 'Malawi cropland');
+leftMap.addLayer(Admin_color, {palette: 'CE33FF'}, 'Malawi_Admin');
+leftMap.addLayer(EPA_color, {palette: '0096FF'}, 'Malawi_EPA', false);
+leftMap.addLayer(TA_color, {palette: 'black'}, 'Malawi_TA', false);
+
+//Visualize the Sentinel 2 high resolution image on the right panel
+var vis = {min: 0.0, max: 0.3, bands: ['B4', 'B3', 'B2']};
+rightMap.addLayer(Sentinel2, vis, 'Sentinel-2');
+rightMap.add(layer2);
+rightMap.addLayer(Admin_color, {palette: 'CE33FF'}, 'Malawi_Admin');
+rightMap.addLayer(EPA_color, {palette: '0096FF'}, 'Malawi_EPA', false);
+rightMap.addLayer(TA_color, {palette: 'black'}, 'Malawi_TA', false);
+
+// Link the two panels
+var linker = ui.Map.Linker([leftMap, rightMap]);
+
+//Ceate a label for the data export
+var downloadUrl = ui.Label('', {
+  stretch: 'horizontal',
+  textAlign: 'left',
+  fontWeight: 'bold',
+  textDecoration: 'underline',
+  fontSize: '10',
+  height: '15px',
+});
+
+//Create a button to download all the data points as a feature collection
+var button = ui.Button('Get link for downloading points',function() {
+  var firstLayer = drawingTools1.layers().get(0);
+  var fc = firstLayer.getEeObject();
+  var sampled = upd_name.sample({
+    region: fc,
+    geometries: true,
+  });
+  //print(sampled);
+  sampled = sampled.map(function(feature) {
+    var point = feature.geometry().coordinates();
+    var long = point.get(0);
+    var lat = point.get(1);
+    return feature.set({
+      longitude: long,
+      latitude: lat,
+    });
+  });
+  
+  //Generate a URL for the points and export as CSV
+  //button.setLabel('Saved!');
+  downloadUrl.setValue('');
+  button.setDisabled(true);
+  sampled.getDownloadURL({
+    filename: 'Malawi_new points',
+    callback: function(url) {
+      downloadUrl.setValue(url);
+      downloadUrl.setUrl(url);
+      button.setDisabled(false);
+    }
+  });
+  
+});
+
+// Create the split panels
+var splitPanel = ui.SplitPanel({
+  firstPanel: leftMap,
+  secondPanel: rightMap,
+  orientation: 'horizontal',
+  wipe: false,
+  style: {stretch: 'both'}
+});
+
+
+// Add captions
+var infoPanel1 = ui.Panel({
+  style: {
+    position: 'bottom-left',
+    padding: '8px 15px',
+    stretch: "both",
+  }
+});
+
+var infoPanel2 = ui.Panel({
+  style: {
+    position: 'bottom-right',
+    padding: '8px 15px',
+    stretch: "both",
+  }
+});
+
+var info1 =ui.Label({
+  value: "Malawi cropland map",
+  style: {
+    fontSize: '12px',
+    margin: '0 0 3px 0',
+    padding: '0'
+  }
+});
+
+var info2 =ui.Label({
+  value: "Malawi Sentinel 2 Image",
+  style: {
+    fontSize: '12px',
+    margin: '0 0 3px 0',
+    padding: '0'
+  }
+});
```

Comment:
These info panels and labels are declared here but they aren't attached together until the end of the script. To me this is a bit confusing and attaching them immediately after their declaration would probably be cleaner. Also by writing the style as a separate variable the code can be much fewer lines.

```javascript
var infoPanel1 = ui.Panel({
  style: {
    position: 'bottom-left',
    padding: '8px 15px',
    stretch: "both",
  }
});

var infoPanel2 = ui.Panel({
  style: {
    position: 'bottom-right',
    padding: '8px 15px',
    stretch: "both",
  }
});

var infoStyle = {fontSize: '12px', margin: '0 0 3px 0', padding: '0'}
var info1 = ui.Label({value: "Malawi cropland map", style: infoStyle});
var info2 = ui.Label({value: "Malawi Sentinel 2 Image", style: infoStyle});
infoPanel1.add(info1);
infoPanel2.add(info2);
```

Reply:
I totally agree with you.